front/src/views/user/history-previous-trips/history-previous-trips.view.js

```diff
@@ -0,0 +1,61 @@
+import { MmtTripCard, MmtTextInput } from '@components';
+
+export default {
+  components: {
+    MmtTripCard,
+    MmtTextInput
+  },
+  data: () => ({
+    searchQuery: '',
+    filter: {
+      year: 0,
+      month: 0
+    }
+  }),
+  computed: {
+    trips () {
+      return this.$store.state.trip.tripsHistory;
+    },
+    years () {
+      return this.trips.map(function (trip) {
+        return new Date(trip.startDate).getFullYear();
+      }).filter(function (trip, index, self) {
+        return self.indexOf(trip) === index;
+      }) || [];
+    },
+    months () {
+      if (this.filter.year) {
+        const lang = this.$store.state.locale.language || 'default';
+        return this.trips.map(function (trip) {
+          return {
+            name: new Date(trip.startDate).toLocaleString(lang, { month: 'long' }),
+            id: new Date(trip.startDate).getMonth() + 1
+          };
+        }).filter(function (trip, index, self) {
+          return self.indexOf(trip) === index;
+        });
+      }
+
```

Comment:
let's use arrow functions. we have limitations only for vue functions like computed, data, hooks etc. others could be `() => {}`

Reply:
ok, I get it